source/SpinalHDL/Data types/bits.rst

```diff
@@ -273,12 +273,13 @@ Misc
      - Return bitcount
      - Int
    * - x.bitsRange
-     - Return the range (x.high downto 0)
+     - Return the range (0 to x.high)
+     - Range
    * - x.valueRange
-     - Return the range (x.minValue downto x.maxValue). Note can't be used for value which overflow the JVM Int capacity.
+     - Return the range of minimum to maximum x values, interpreted as an unsigned integer (0 to 2 \*\* width - 1).
```

Comment:
Three things that come to my mind here: 
1. it would be nice to keep range descriptors that are similar to the code "(x.minValue downto x.maxValue)"
2. Usually I'd expect "HI downto LO " ranges instead of "LO to HI"
3. The previous comment had this information about the overflow of the JVM Int capacity. Shouldnt that be kept as a note?

Reply:
ad 1: I wanted to avoid that since it's not a numeric type that we are talking about
ad 2: that not how it's implemented, a range of low to high is returned
ad 3: That comment is still there, just as a not at the end of the table, like the other caveats that are described on the page

Reply:
For the first point, the notion of "range" is [by definition](https://en.wikipedia.org/wiki/Interval_(mathematics)) about real numbers: there must be an axis / ordering relationship so in a way, "range" can exist only while considering Bits as real numbers, and in this case UInt (it could be SInt or even floating point numbers).

So IMO there is an issue in the API about the notion of range for Bits; and as far as this function exists, explicitly stating that this specific function is, in a way, casting Bits to UInt, is not an issue.